### DISCUSSION TITLE
pagination and enum listing

### DISCUSSION THREAD

**Discussion by mychatterbox** [mrmcc3/vibing-dev#20] [Q&A]
Your blog is really cool.
Since I am not a programmer, it was not easy to enable .md files, and the pagination function is really difficult to implement.
Will pagination be added at some point?
One thing to add is that when I click on enum (Article, Note, etc), I want to list only the relevant items, but it is not easy. Don't you need this feature?

**Comment by mrmcc3** [comment]
Hey thanks!

I decided to use [markdoc](https://markdoc.dev/docs/overview) because I've found it easier to work with and configure then markdown/MDX. It's a superset of markdown so you can write vanilla markdown in `.mdoc` files and it will just work.

I've found it nice that you can extend markdoc to support custom astro components for example if you use

```md
{% box %}

this will be in a blue box using `src/components/box.astro`

{% /box %}
```

Pagination is always a bit of a challenge to implement and usually isn't worth the effort IMO. I actually find it kind of frustrating If I'm on someone's site and it only shows a handful of links. I'd prefer if i can just see all the articles instead of hunting down an archive link / or clicking through pages. Everything is more accessible. I guess if I had 100s of articles it might be worth it but I don't see myself having that problem anytime soon 😅 . https://overreacted.io/ is a popular blog that does this

> One thing to add is that when I click on enum (Article, Note, etc), I want to list only the relevant items, but it is not easy. Don't you need this feature?

This one shouldn't be to hard. You could do it two ways: client side so the page doesn't change or on a separate page. For the separate page - off the top of my head 
- add the page[s] to show the filtered links `src/pages/kind/[kind].astro`
- export `getStaticPaths` that looks up the published collection and returns each `kind` in use as props.
- then when rendering the page use specific `kind` to filter the published collection and display them (similar to how index.astro filters by draft)

If you're interested in how it's done I could add it.

**Reply by mychatterbox** [reply]
After reading your reply and thinking about it, I realized.
It's difficult to imagine when my blog will have over 100 posts. 1 year? 10 years? 

Obviously, it's okay to not have Pagination, enum and tag features right now. 
Of course, it would be even better if there was a feature that really matches the current template!
(This may not be a secret, Northeast Asians like me are familiar with the categories.)
It's great fun to see what features are being added to this great template.
In the meantime, I'll look at the astro doc a little more.

Did you find my sentences a little awkward? It's difficult to convey my feelings accurately with just Google Translate.

**Reply by mrmcc3** [reply]
I [added](https://github.com/mrmcc3/vibing-dev/commit/0d4d90e857c9fcae979cca1c0ed33f007baf85e8#diff-d61071dacb80817ebcbbf6297d0bac4585f7a11d462cfaf54a5040b4d957ac05) pages filtered by kind if you're interested. 

> Did you find my sentences a little awkward?

No I understood fine. DW I have that problem and english is my first language 😅 

Good luck with your astro sites!
